packages/vm/src/runBlock.ts

```diff
@@ -125,26 +125,39 @@ export async function runBlock(this: VM, opts: RunBlockOpts): Promise<RunBlockRe
     if (this.DEBUG) {
       debug(`Set provided state root ${bytesToHex(root)} clearCache=${clearCache}`)
     }
-    await state.setStateRoot(root, clearCache)
+    await stateManager.setStateRoot(root, clearCache)
   }
 
   if (this.common.isActivatedEIP(6800)) {
-    if (!(state instanceof StatelessVerkleStateManager)) {
+    if (!(stateManager instanceof StatelessVerkleStateManager)) {
       throw Error(`StatelessVerkleStateManager needed for execution of verkle blocks`)
     }
     if (this.DEBUG) {
       debug(`Initializing StatelessVerkleStateManager executionWitness`)
     }
     if (clearCache) {
-      ;(this._opts.stateManager as StatelessVerkleStateManager).clearCaches()
+      stateManager.clearCaches()
     }
 
-    ;(this._opts.stateManager as StatelessVerkleStateManager).initVerkleExecutionWitness(
-      block.header.number,
-      block.executionWitness
-    )
+    // Update the stateRoot cache
+    await stateManager.setStateRoot(block.header.stateRoot)
+
+    // Populate the execution witness
+    stateManager.initVerkleExecutionWitness(block.header.number, block.executionWitness)
+
```

Comment:
Can we move this below the error condition checks or do they need to happen before we check for parent stateRoot?

Reply:
Good point, I moved this up